### ISSUE TITLE
Incorrect "Syntax Error: attributes are not allowed on IF_EXPR"

### ISSUE THREAD

**Issue by bjorn3** [rust-analyzer/rust-analyzer#4386]
```rust
#[cfg(false_)]
if true {}
```

rustc accepts this. Also the syntax error is reported on the closing `}`, not the `#[cfg(false_)]` or `if`.

**Comment by d4h0** [comment]
The following code triggers the incorrect systax error as well:

```rust
    if original_name == ident_name.to_string() {
        None
    } else {
        Some(quote! {#[serde(rename = #original_name)]})
    }
```

**Comment by xobs** [comment]
I'm also running into this problem as well because I can't otherwise disable a `clippy` lint, and this is apparently the official way to disable lints:

https://github.com/rust-lang/rust-clippy/issues/5728

**Comment by matklad** [comment]
Both examples work on master now!

**Comment by bolinfest** [comment]
I'm running rust-analyzer nightly right now (6824cf4) in VS Code 1.47.1 and it is giving me `Syntax Error: attributes are not allowed on IF_EXPR` on this code:

```rust
#[cfg(unix)]
if let Some(raw_os_error) = error_from_execv.raw_os_error() {
```

**Comment by bolinfest** [comment]
Hmm, sorry, I think I also had an old version of the rust-analyzer extension installed (I may have built it from source and given it a different name) because I was also seeing issues about the `rust-analyzer.extendSelection` command being missing, which was removed earlier this year: https://github.com/rust-analyzer/rust-analyzer/commit/13eddd7c499388e956e47aae6a7210e43eb40d55.

Now that I removed that extension from VS Code, things appear to be working as intended!
